pyramid_hypernova/context_manager.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from contextlib import contextmanager
+
+from pyramid_hypernova.rendering import RenderToken
+from pyramid_hypernova.tweens import configure_hypernova_batch
+
+
+@contextmanager
+def hypernova_batch(request, registry):
+    """A context manager that performs hypernova token replacement in a batch.
+    Write the content you wish to modify in body['content'] where body is the
+    yielded object.
+
+    :param request: a Pyramid request object
+    :type request: pyramid.util.Request
+    :param registry: a Pyramid application registry object
```

Comment:
Similarly, this context manager doesn't really need to know about the Pyramid registry object. It only needs to be passed in a `batch_request_factory` method (which could be initialized via the Pyramid registry:

```
@contextmanager
def hypernova_batch(batch_request_factory):
    ...

with hypernova_batch(lambda: configure_hypernova_batch(registry)) as hypernova_batch, body:
   ...
```

At that point, the `hypernova_batch` ctx manager is totally unaware of Pyramid, and could be used outside of a Pyramid request.

Reply:
I made the context manager be passed in a hypernova batch object since `batch_request_factory` method would also need its arguments (built from registry) passed in. Let me know if this looks good.